superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -1013,6 +1156,41 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
     );
   };
 
+  if (file.length > 0 && importingModel) {
```

Comment:
can we refactor the logic not to use `importingModel` and instead leverage `file`, `confirmedOverwrite`, and `passwords` to dictate the state and what the modal should show.

Reply:
Can do! Changed in [`this commit`](https://github.com/apache/superset/pull/19314/commits/a5d4761a49da4c4799022ac437698431d5b267f3).